src/Menu/Menu.js

```diff
@@ -0,0 +1,232 @@
+    }
+  }
+
+
+  // Internal functions
+
+  open() {
+    this.props.onOpen && this.props.onOpen();
+
+    this.setState({
+      expanded: true
+    });
+
+    document.addEventListener('click', this.onDocumentClick, true);
+  }
+
+  close() {
+    this.props.onClose && this.props.onClose();
+
+    this.setState({
+      expanded: false
+    });
+
+    document.addEventListener('click', this.onDocumentClick, true);
+  }
+
+  toggle() {
+    this.state.expanded ? this.close() : this.open();
+  }
+
+  getFocusableElements() {
+    return this.refs.menu.querySelectorAll('button:not([disabled]), [href]:not([disabled]), input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([tabindex="-1"]):not([disabled])');
+  }
+
+
+  // Event handlers
+
+  onDocumentClick(e) {
+    if (!this.props.closeOnDocumentClick) return;
+    if (this.refs.menu && (e.target === this.refs.menu || this.refs.menu.contains(e.target))) return;
+
+    this.close();
+  }
+
+  onTriggerClick(e) {
+    if (this.state.expanded && this.props.triggerDestination) {
+      return; // do nothing. let the browser follow the link
+    }
+
+    e.preventDefault();
+
+    this.toggle();
+  }
+
+  onCloseClick(e) {
+    this.getFocusableElements()[0].focus();
+    this.close();
+  }
+
+  onKeyDown(e) {
+    if (!this.state.expanded) return;
+    switch(e.key) {
+      case 'Escape':
+        e.preventDefault();
+        e.stopPropagation();
+        
+        this.getFocusableElements()[0].focus();
+        this.close();
+        break;
+      case 'Enter':
+        // Using focusable elements instead of a ref to the trigger
+        // because Hyperlink and Button can handle refs as functional compoenents
+        if (document.activeElement == this.getFocusableElements()[0]) {
+          e.preventDefault();
+          this.toggle();
+        }
+        break;
+      case 'Tab':
+        // Trap focus
+        const focusableElements = Array.from(this.getFocusableElements());
+        const indexOfActiveElement = focusableElements.indexOf(document.activeElement);
+
+        if (indexOfActiveElement === focusableElements.length - 1 && !e.shiftKey) { // last. cycle forward
+          e.preventDefault();
+          focusableElements[0].focus();
+        }
+
+        if (indexOfActiveElement === 0 && e.shiftKey) { // first. cycle backward
+          e.preventDefault();
+          focusableElements[focusableElements.length - 1].focus();          
+        }
+        break;
+    }
+  }
+
+  onMouseEnter(e) {
+    if (!this.props.respondToPointerEvents) return;
+
+    this.open();
+  }
+
+  onMouseLeave(e) {
+    if (!this.props.respondToPointerEvents) return;
+
+    this.close();
+  }
+
+
+
+
+  render() {
+    const triggerContent = this.state.expanded && this.props.triggerExpandedContent ? 
+      this.props.triggerExpandedContent : 
+      this.props.triggerContent;
+    
+    const triggerProps = {
+      className: classNames("menu-trigger", {
```

Comment:
Styling: Per the edX JavaScript style guide (=== AirBnb style guide), looks like we want single quotes throughout.  https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/javascript-guidelines.html

Reply:
Go through https://github.com/edx/frontend-cookiecutter-library for eslint and other things.